src/deluge/modulation/lfo.cpp

```diff
@@ -22,3 +24,24 @@ uint32_t getLFOInitialPhaseForZero(LFOType waveType) {
 		return 0;
 	}
 }
+
+void LFO::setInitialPhase(const LFOConfig& config) {
+	// Historically LFO1 didn't do much initial phase setting unless it was sync'ed.
+	// LFO2 on the other hand didn't support sync, and always went for the negative
+	// extreme regardless of wave type.
+	//
+	// This logic matches the historical behaviour for synced LFO1 and unsynced LFO2,
+	// and will make synced LFO2 work the same as synced LFO1.
+	//
+	// TODO: Despite losing historical behaviour, consistent initial phase behaviour
```

Comment:
starting from negative extreme in the saw makes it more usable for lots of note modulation since it guarantees that the modulation will be active at note on. It'll be especially true if we add one shot LFOs

If we remove it by default we should make it configurable in an LFO phase menu or something

Reply:
I was a little unclear.

Currently:
- LFO1 unsynced: ???
- LFO1 synced: from zero for sine and triangle, from negative extreme for others.
- LFO2 always: from negative extreme.

With this patch:
- LFO2 unsynced: from negative extreme.
- LFO1 synced: from zero for sine and triangle, from negative extreme for others.

My comment refers to using the sync logic for unsynced LFOs as well: starting from zero for sine and triangle, and keeping saw and others starting from negative.

If starting everything from negative for LFO2 makes it more usable, that's even simpler. It's the "do one thing for synced and another for unsynced" that seems a bit messy: I just wanted to match LFO1 behaviour for sync, assuming there was a good reason for it, without changing historical unsynced behaviour for LFO2.

Reply:
I'd lean towards always starting from negative so that they're all consistent, then changing saw to sine doesn't change the start of the note

Reply:
Keep global initial phases as they are, or go to negative there as well?

Reply:
Probably leave them? I'm sure somebody is relying on the current behaviour